app/views/licences/tabs/contact-details.njk

```diff
@@ -4,14 +4,18 @@
   <p><a href='/customer/{{ customerId }}/#contacts'>Go to customer contacts</a></p>
 {% endif %}
 
+{% if not customerContacts.length > 0  and not licenceContacts.length > 0 %}
+  <p>No contacts found.</p>
+{% endif %}
```

Comment:
This one is really a matter of opinion (so feel free to 🥊 me!) But I think it reads clearer as

```suggestion
{% if customerContacts.length == 0 and licenceContacts.length == 0 %}
  <p>No contacts found.</p>
{% endif %}
```

🤷

Reply:
If it passes the test no probs.

Reply:
I can see that this has caused the test to fail. However, I believe it's the test that needs fixing not the view.

Assuming I've followed and understood what is happening correctly; `ViewLicenceContactDetails` will _always_ return

```javascript
  // ...
  customerContacts: [],
  licenceContacts: []
}
```

It is just whether they are populated or not. However, when we stub `ViewLicenceContactDetailsService` in the "when a request is valid and has no contact details" test we are only returning `{ activeTab: 'contact-details' }` which doesn't reflect what would actually happen.

If we update the stub to be

```javascript
Sinon.stub(ViewLicenceContactDetailsService, 'go').resolves({
  activeTab: 'contact-details',
  customerContacts: [],
  licenceContacts: []
})
```

then the test will pass again.

Reply:
soddy test that, poor craftmanship. 

have updated and fixed.

Reply:
unresolved. 

@Cruikshanks resolve if you are happy.